### ISSUE TITLE
Wrong hint

### ISSUE THREAD

**Issue by sobhanattar** [kalessil/phpinspectionsea#234]
Consider following code:

```php
if (is_numeric($companyBaseScoreInfo->time_range) &&
    $companyBaseScoreInfo->time_range !== ''
) {
    $companyBaseScoreInfo->time_range = 'some number';
}
else {
    $companyBaseScoreInfo->time_range = '';
}
```
and the help it gave me:
```Assignment in this branch should be moved before the if. less... (Ctrl+F1) 
Reports conditional constructions where default value can be moved 
from the 'else' part to right before the conditional. 
It can simplify readability and make supporting code easier.
if () {
    $var = [expression];
}
 [elseif () { ... }]
else {
    $var = [array|string|constant|number];
}
```

which is obvious that is a destroy the code, because if I give the `$companyBaseScoreInfo->time_range = ''` before `if` then the condition will always be `false`.

Do I miss something or the helper really wants to give my app some help?

**Comment by kalessil** [comment]
@sobhanattar, you are right here - the inspection is not checking conditions and this needs to be fixed.
Thank you for reporting.

**Comment by kalessil** [comment]
Fixed, fix will be available with the next release (ETA 2 weeks)

**Comment by sobhanattar** [comment]
Thank you

**Comment by kalessil** [comment]
You are welcome )
